pkg/apis/testharness/v1beta1/test_types.go

```diff
@@ -13,6 +13,19 @@ type RestConfig struct {
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
+// TestFile contains attributes of a single test file.
+type TestFile struct {
+	// The type meta object, should always be a GVK of kuttl.dev/v1beta1/TestFile.
+	metav1.TypeMeta `json:",inline"`
+	// Set labels or the test suite name.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Which test runs should this file be used in. Empty selector matches all test runs.
+	TestRunSelector *metav1.LabelSelector `json:"testRunSelector,omitempty"`
```

Comment:
This particular arrangement of which side labels vs selectors go was @eddycharly 's idea, so maybe he can chime in on where it came from. Maybe it's following some pattern seen elsewhere?

I guess you can think about this in two ways:
1. (_this implementation_): a "test run" (i.e. a `kuttl` invocation) comes with a set of labels attached. These labels represent some characteristics of this test invocation. In my use case, there would would be a single label with just two possible values:`openshift=true` or `openshift=false`, but you could imagine multiple labels and values for a more elaborate test matrix (such as the cartesian product of `plaform={openshift,vanilla,eks},k8s-version={1.28,1.29,1.30}`). Then individual files in the test suite provide an optional label selector, which needs to match the label set in order for the file to be considered for the test run. Obviously the selector can contain a subset of label keys that the test run has. For example a given file might only select one or more `k8s-version` values and be neutral w.r.t. `platform`. The notable edge case is a file that (be default) has no selector attached - an empty selector matches all possible label sets, meaning this feature is fully backwards-compatible - a file without a `TestFile` is included in all test invocations, just like before this change.
2. (I haven't thought about this case earlier so I'm inventing things as I go.) Each _test file_ comes with an optional _label set_ (empty by default). Let's say that some file tests some API introduced in version 1.29 but also present in 1.30. Since a label set can only specify a single value for a given label key, we'd either have to allow multiple label sets per file (eek!) or introduce a copy of the file per k8s version (ouch) or instead of labeling by k8s version, label by feature availability `feature-foo={true,false}`. Then, each _test run_ can have a _label selector_ attached, and this selector would need to select the APIs supported. This could quickly create an explosion of labels and combinations. What's worse, files which are agnostic about a given API would have to somehow advertise that fact, in order to be selected by a selector which does mention a given API.

So it looks _to me_ that only solution 1 is in fact realistic.